### PR TITLE
Feature/aql performance improvements

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Added traversal option `edgeCollections` to restrict the traversal to certain edge collections when using named graphs
Additionally, speed up internal selection of appropriate AqlItemBlock bucket by using clz. 
Plus, remove the traversal's vertex output register from the register plan in case it is not needed.

- [x] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This PR adds tests that were used to verify all changes:

- [x] Added new **integration tests** (i.e. in shell_server / shell_server_aql)

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8721/

### CODE REVIEW

File: arangod/Aql/AqlItemBlockManager.cpp [hunk 1]
```diff
@@ -185,35 +185,20 @@ size_t AqlItemBlockManager::Bucket::getId(size_t targetSize) noexcept {
...
+#elif defined(_MSC_VER)
+  unsigned long index;
+  _BitScanReverse64(&idx, static_cast<__int64>(targetSize));
+  size_t value = static_cast<size_t>(index);
+#else
+  static_assert(false, "no known way of computing clz");
+#endif
+
+  TRI_ASSERT(value < numBuckets);
+  return value;
```
Comment (danielhlarkin) [inline arangod/Aql/AqlItemBlockManager.cpp]: Can we extract this whole clz logic out into a method in lib/Basics somewhere and hide the compiler-specifics in the .cpp there? It's probably useful elsewhere in our code, and would be nice to re-use without reinventing the wheel every time. Actually, there's a good chance we already use something similar somewhere...

Comment (jsteemann) [reply arangod/Aql/AqlItemBlockManager.cpp]: Ok, just moved this into NumberUtils.cpp

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8729/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8733/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8755/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8761/

**Comment by jsteemann** [comment]
Tests blue

**Pull request merged**